pkg/didcomm/common/messenger/messenger.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package messenger
+
+// The messenger package is responsible for the handling of communication between agents.
+// It takes care of threading, timing, etc.
+// Each message that we are going to send should be map[string]interface{} type.
+// e.g we have type model.Ack{Type: "type", ID: "id"}
+// it should be converted to: map[@id:id @type:type]
+// NOTE: The package modifies message data by JSON tag name according to aries-rfcs.
+//       Fields like @id, ~thread , etc. are redundant and may be rewritten.
+
+// Client provides methods for the communication
+type Client interface {
```

Comment:
Who are the intended consumers of this interface?

Reply:
Any service/protocol may use it. `aries.context` will provide it. We will be able to get a messenger from the context.